diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 3f32b34264..8fb1f40ea0 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -333,6 +333,7 @@ dtb-$(CONFIG_ARCH_SOCFPGA) +=				\
 
 dtb-$(CONFIG_TARGET_SOCFPGA_CL_MITYSOM_5CS) +=		\
 	socfpga_mitysom5cse-l2-3y8_devkit.dtb		\
+	socfpga_mitysom5cse-l2-3y8_xa.dtb		\
 	socfpga_mitysom5cse-h4-3ya_devkit.dtb		\
 	socfpga_mitysom5cse-h4-8ya_devkit.dtb		\
 	socfpga_mitysom5csx-h5-4ya_devkit.dtb		\
diff --git a/arch/arm/dts/socfpga_mitysom5cse-l2-3y8_xa-u-boot.dtsi b/arch/arm/dts/socfpga_mitysom5cse-l2-3y8_xa-u-boot.dtsi
new file mode 100644
index 0000000000..1521f669fe
--- /dev/null
+++ b/arch/arm/dts/socfpga_mitysom5cse-l2-3y8_xa-u-boot.dtsi
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * U-Boot additions
+ *
+ * Copyright (C) 2012 Altera Corporation <www.altera.com>
+ * Copyright (c) 2018 Simon Goldschmidt
+ * Copyright (C) 2020 Critical Link LLC <www.criticallink.com>
+ */
+
+/*
+ * This dtsi is automagically included by the dts with the matching prefix
+ * (before "-u-boot.dtsi"). It pulls in our U-Boot-specific changes so the
+ * main dts can stay synced with its Linux counterpart.
+ */
+
+#include "socfpga_mitysom5cs-u-boot.dtsi"
+
+/{
+	/* With CONFIG_DISPLAY_BOARDINFO or CONFIG_DISPLAY_BOARDINFO_LATE, this
+	 * model string is displayed in the console */
+	model = "MitySOM-5CSE-L2-3Y8 Altera SOCFPGA Cyclone V";
+};
diff --git a/arch/arm/dts/socfpga_mitysom5cse-l2-3y8_xa.dts b/arch/arm/dts/socfpga_mitysom5cse-l2-3y8_xa.dts
new file mode 100644
index 0000000000..bde8c01b08
--- /dev/null
+++ b/arch/arm/dts/socfpga_mitysom5cse-l2-3y8_xa.dts
@@ -0,0 +1,53 @@
+#include "socfpga_mitysom5cse.dtsi"
+#include <dt-bindings/interrupt-controller/irq.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	aliases {
+		/* this allow the ethaddr uboot environmnet variable contents
+		 * to be added to the gmac1 device tree blob.
+		 */
+		ethernet0 = &gmac1;
+	};
+
+};
+
+&i2c1 {
+	speed-mode = <0>;
+	status = "okay";
+    /* Power Monitor */
+	ltc2945@6F {
+		compatible = "ltc2945";
+		reg = <0x6F>;
+    };
+};
+
+&mmc {
+	status = "okay";
+	num-slots = <1>;
+	broken-cd;
+	cap-mmc-highspeed;
+	cap-sd-highspeed;
+	altr,dw-mshc-ciu-div = <3>;
+	altr,dw-mshc-sdr-timing = <0 3>;
+	bus-width = <4>;
+};
+
+&gmac1 {
+	status = "okay";
+	phy-mode = "rgmii";
+
+	phy-handle = <&phy1>;
+	max-frame-size = <3800>;
+
+	snps,reset-delays-us = <0 10000 1000000>;
+	snps,reset-gpio = <&porta 28 GPIO_ACTIVE_LOW>;
+	snps,reset-active-low;
+	mdio0 {
+		compatible = "snps,dwmac-mdio";
+		phy1: ethernet-phy@0 {
+			reg = <0>;
+		};
+	};
+
+};
